examples/image_classification-tf.ipynb

```diff
@@ -0,0 +1,7883 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #4.        columns=["image", "label"],
</pre><p> </p><p>What is the significance of <code>columns</code>? Each sample of <code>train_ds</code> and <code>val_ds</code> has two keys: <code>pixel_values</code> and <code>label</code>.Will <code>to_tf_dataset</code> convert the <code>pixel_values</code> key to <code>image</code>? Or how is it?           </p><p> </p><p>Or better yet -- take a batch from <code>train_set</code> and investigate it (like printing the shape and so on).  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/notebooks/pull/202/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='54'/>

Reply:
<p><code>columns</code> specifies the columns to be read into the <code>tf.data.Dataset</code> object being created. (c.f. the <a href="https://huggingface.co/docs/datasets/v2.3.2/en/package_reference/main_classes#datasets.Dataset.to_tf_dataset" target="_blank">datasets</a> docs). <code>train_ds</code>  and <code>val_ds</code> don't have the <code>pixel_values</code>  key yet - they have <code>label</code> and <code>image </code> . <code>pixel_values</code> feature is added in the <code>preprocess_xxx</code> function.</p><p> </p><p> I'll add a cell below taking a batch and showing its contents 👍</p>

Reply:
<p>Okay. </p><p> </p><p>Sorry for another follow-up question. If <code>image</code> and <code>label</code> are being loaded in the <code>tf.data.Dataset</code> object created how <code>pixel_values</code> is being used?     </p>

Reply:
<p>I would also have thought you'd need to put <code>pixel_values</code> in <code>columns</code>! I don't know why it isn't being dropped - let me investigate that.</p>

Reply:
Update (cc @amyeroberts): `pixel_values` should not be in the dataset because you didn't request it in `columns` or `label_cols`. It's only there because of an oversight in the recent `to_tf_dataset` code - it survives because it's added by the dataset transformation, and isn't a column in the original pre-transformation dataset. You definitely don't want to rely on that behaviour, though - `columns` and `label_cols` should contain the names of all the columns you want to see in the output!